### PR TITLE
Deprecate `describeEnum` + type stricter `EnumProperty` 

### PR DESCRIPTION
**Pull request by bernaferrari**
Sequel to https://github.com/flutter/website/pull/8571 and https://github.com/flutter/flutter/pull/125016. This is super small, but broke a test that copied a Flutter file (and was not modified when Flutter got updated).

### CODE REVIEW

File: src/release/breaking-changes/enum-property-type-stricter.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Dart 2.17 introduced enhanced enums. With them, Enum became a type.
+Before that, it was common to use classes as enums, and `EnumProperty`
+was used to represent them.
+
+Since `describeEnum` is being deprecated, `EnumProperty` is being updated to
+be type stricter.
+
+## Description of change
+
+`EnumProperty` expects null or an enum.
```
Comment (sfshaza2) [inline src/release/breaking-changes/enum-property-type-stricter.md:23]:
```suggestion
The `EnumProperty` now expects null or an `Enum`;
you can no longer pass it a non-`Enum` class.
```

File: src/release/breaking-changes/index.md [hunk 2]
```diff
@@ -46,6 +48,8 @@ release, and listed in alphabetical order:
 [Deprecated `RouteInformation.location`]: {{site.url}}/release/breaking-changes/route-information-uri
 [Customize tabs alignment using the new `TabBar.tabAlignment` property]: {{site.url}}/release/breaking-changes/tab-alignment
 [Updated EditableText scroll into view behavior]: {{site.url}}/release/breaking-changes/editable-text-scroll-into-view
+[Deprecate describeEnum]: {{site.url}}/release/breaking-changes/describe-enum
+[Make EnumProperty be type stricter]: {{site.url}}/release/breaking-changes/enum-property-type-stricter
```
Comment (sfshaza2) [inline src/release/breaking-changes/index.md:52]: Wait! Wait a minute. I never saw this "Deprecate describeEnum" PR! I would never have approved this. Why can't these two breaking changes be merged into one breaking change? Oy, we have a policy where we never approve breaking change PRs that don't have a "landed in" field filled out. I must have been on vacation when this one happened.

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
I have some rather large changes. Also, the previous "Deprecate describeEnum" PR has some of the same issues.

**Review by atsansone — APPROVED** [review]

File: src/release/breaking-changes/index.md (hunk 2 above)
Comment (bernaferrari) [reply src/release/breaking-changes/index.md:52]: They can. Someone approved that (probably by mistake) and there were some adjustments to make, like this property that appeared later. I'll merge them.

Comment (bernaferrari) [reply src/release/breaking-changes/index.md:52]: I don't want to incriminate anyone, but you merged it lol

**Comment by bernaferrari** [comment]
I merged the older `describeEnum` document with this method (so it is a single document). Tell me how you feel.

**Comment by bernaferrari** [comment]
Friendly bump.

**Comment by bernaferrari** [comment]
cc @sfshaza2

File: src/release/breaking-changes/describe-enum.md [hunk 3]
```diff
@@ -1,40 +1,51 @@
...
 of `describeEnum(Enum.something)` should use
 `Enum.something.name` instead.
 
+The class EnumProperty was modified to extend `<T extends Enum?>` instead of `<T>`. Existing uses of `EnumProperty<NotAnEnum>` should use `DiagnosticsProperty<NotAnEnum>` instead.
+
 ## Context
 
-Dart 2.17 introduced enhanced enums. With them, Enum became a type
-and all enums got a `name` getter, which made `describeEnum` redundant.
+Dart 2.17 introduced [enhanced enums][], which added `Enum` as a type. As a result, all enums got a `name` getter, which made `describeEnum` redundant. Before that, enum classes were often diagnosticated using an `EnumProperty`.
```
Comment (sfshaza2) [inline src/release/breaking-changes/describe-enum.md:16]: I'm so tempted to leave diagnosticated in, but we _should_ replace it something like, tested, evaluated, analyzed, investigated...

```suggestion
Dart 2.17 introduced [enhanced enums][], which added `Enum` as a type.
As a result, all enums got a `name` getter, which made `describeEnum`
redundant. Before that, enum classes were often
analyzed using an `EnumProperty`.
```

File: src/release/breaking-changes/describe-enum.md [hunk 4]
```diff
@@ -1,40 +1,51 @@
...
 - Replace `describeEnum(Enum.something)` with `Enum.something.name`.
 
+The `EnumProperty` now expects null or an `Enum`;
+you can no longer pass it a non-`Enum` class.
+
 ## Migration guide
 
-If you used `describeEnum(Enum.field)` to access the string value from an
-enum, you can now call `Enum.field.name`.
+If previously used `describeEnum(Enum.field)` to access the string value from
```
Comment (sfshaza2) [inline src/release/breaking-changes/describe-enum.md:35]:
```suggestion
If you previously used `describeEnum(Enum.field)` to access the string value from
```

File: src/release/breaking-changes/describe-enum.md [hunk 5]
```diff
@@ -56,21 +70,28 @@ API documentation:
...
 [`describeEnum` stable]: {{site.api}}/flutter/lib/src/foundation/describeEnum.html
 
+[`EnumProperty` stable]: {{site.api}}/flutter/lib/src/foundation/EnumProperty.html
+
 <!-- Master channel link: -->
 {% include docs/master-api.md %}
 
 [`describeEnum` main]: {{site.master-api}}/flutter/lib/src/foundation/describeEnum.html
 
-[☂️ Cleanup SemanticsFlag and SemanticsAction issue]: {{site.repo.flutter}}/issues/123346
```
Comment (sfshaza2) [inline src/release/breaking-changes/describe-enum.md:75]: Why did you delete this line? It should be:

```
- [Cleanup SemanticsFlag and SemanticsAction issue][cleanup-issue]
```

And then [cleanup-issue] is defined below. It allows a shortcut to that loooong title.

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Sorry I missed your updates before. This looks much better, but there are a few little things, still.

File: src/release/breaking-changes/describe-enum.md (hunk 5 above)
Comment (bernaferrari) [reply src/release/breaking-changes/describe-enum.md:75]: I don't understand but I think I did this now. I put the long title reference in the related issues, and in the master link it links to the shorter title.

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Comment by bernaferrari** [comment]
Is the failing check intended (will be auto fixed once the original thing is merged)?

File: src/release/breaking-changes/index.md [hunk 6]
```diff
@@ -48,6 +49,7 @@ release, and listed in alphabetical order:
 [Customize tabs alignment using the new `TabBar.tabAlignment` property]: {{site.url}}/release/breaking-changes/tab-alignment
 [Updated EditableText scroll into view behavior]: {{site.url}}/release/breaking-changes/editable-text-scroll-into-view
 [Deprecate `textScaleFactor` in favor of `TextScaler`]: {{site.url}}/release/breaking-changes/deprecate-textscalefactor
+[Deprecate describeEnum and update EnumProperty to be type strict]: {{site.url}}/release/breaking-changes/enum-property-type-stricter
```
Comment (sfshaza2) [inline src/release/breaking-changes/index.md:52]:
```suggestion
[Deprecate describeEnum and update EnumProperty to be type strict]: {{site.url}}/release/breaking-changes/describe-enum
```

Comment (bernaferrari) [reply src/release/breaking-changes/index.md]: That was dumb, thanks a lot!!

**Comment by bernaferrari** [comment]
(do I still need the review form @parlough ? It's been a while I don't break anything, I forgot how the process is).
